x-pack/plugins/spaces/server/saved_objects/spaces_saved_objects_client.ts

```diff
@@ -164,19 +166,32 @@ export class SpacesSavedObjectsClient implements SavedObjectsClientContract {
     let namespaces = options.namespaces;
     if (namespaces) {
       const spacesClient = await this.getSpacesClient;
-      const availableSpaces = await spacesClient.getAll('findSavedObjects');
-      if (namespaces.includes('*')) {
-        namespaces = availableSpaces.map((space) => space.id);
-      } else {
-        namespaces = namespaces.filter((namespace) =>
-          availableSpaces.some((space) => space.id === namespace)
-        );
-      }
-      // This forbidden error allows this scenario to be consistent
-      // with the way the SpacesClient behaves when no spaces are authorized
-      // there.
-      if (namespaces.length === 0) {
-        throw this.errors.decorateForbiddenError(new Error());
+      const emptyResult = Promise.resolve({
+        page: options.page ?? 1,
+        per_page: options.perPage ?? 20,
+        total: 0,
+        saved_objects: [],
+      } as SavedObjectsFindResponse<T>);
+
+      try {
+        const availableSpaces = await spacesClient.getAll('findSavedObjects');
+        if (namespaces.includes('*')) {
+          namespaces = availableSpaces.map((space) => space.id);
+        } else {
+          namespaces = namespaces.filter((namespace) =>
+            availableSpaces.some((space) => space.id === namespace)
+          );
+        }
+        if (namespaces.length === 0) {
+          // return empty result, since the user is unauthorized in this space (or these spaces), but we don't return forbidden errors for `find` operations
+          return emptyResult;
+        }
+      } catch (err) {
+        if (Boom.isBoom(err) && err.output.payload.statusCode === 403) {
```

Comment:
😬 I really, _really_ want to get rid of this forbidden error that the `SpacesClient` throws. @watson has a draft PR to investigate if this is feasible.

This is fine for now, but let's collectively try to remember this if/when we get rid of that from the `SpacesClient`.

Reply:
OK, in that case I'll put a comment in `SpacesClient` so we don't forget 👍